app/controllers/Application.scala

```diff
@@ -922,7 +925,23 @@ object Application extends Controller with Security {
   }
 
   def currentSubscriberId(implicit request: play.api.mvc.RequestHeader) = {
-    request.session.get("subscriber").getOrElse("0").toInt
```

Comment:
I ended up not refactoring this even though it is a bit clunky. After thinking about it, I decided it was very clear what was happening even if it wasn't very slick. That said, alternate solutions are welcome :)